src/main/java/frc/robot/Subsystems/DriveSubsystem.java

```diff
@@ -0,0 +1,233 @@
+      Rotation2d.fromDegrees(-m_gyro.getAngle()),
+      new SwerveModulePosition[] {
+          m_frontLeft.getPosition(),
+          m_frontRight.getPosition(),
+          m_rearLeft.getPosition(),
+          m_rearRight.getPosition()
+      });
+
+  /** Creates a new DriveSubsystem. */
+  public DriveSubsystem() {
+
+    powerDistribution.clearStickyFaults();
+    // SmartDashboard.putNumber("heading", m_gyro.getAngle());
+  }
+
+  @Override
+  public void periodic() {
+    // Update the odometry in the periodic block
+    m_odometry.update(
+        Rotation2d.fromDegrees(-m_gyro.getAngle()),
+        new SwerveModulePosition[] {
+            m_frontLeft.getPosition(),
+            m_frontRight.getPosition(),
+            m_rearLeft.getPosition(),
+            m_rearRight.getPosition()
+        });
+      
+    double ang = m_gyro.getRotation2d().getDegrees();
+
+    SmartDashboard.putNumber("delta heading", ang - prevAngle);
+
+    prevAngle = ang;
+    SmartDashboard.putNumber("heading", ang);
+
+    SmartDashboard.putNumber("right stick angle", rightAngGoal);
+    SmartDashboard.putNumber("turn direction", turnDir);
+  }
+
+  /**
+   * Returns the currently-estimated pose of the robot.
+   *
+   * @return The pose.
+   */
+  public Pose2d getPose() {
+    return m_odometry.getPoseMeters();
+  }
+
+  /**
+   * Resets the odometry to the specified pose.
+   *
+   * @param pose The pose to which to set the odometry.
+   */
+  public void resetOdometry(Pose2d pose) {
+    m_odometry.resetPosition(
+        Rotation2d.fromDegrees(-m_gyro.getAngle()),
+        new SwerveModulePosition[] {
+            m_frontLeft.getPosition(),
+            m_frontRight.getPosition(),
+            m_rearLeft.getPosition(),
+            m_rearRight.getPosition()
+        },
+        pose);
+  }
+
+  /**
+   * Method to drive the robot using joystick info.
+   *
+   * @param xSpeed        Speed of the robot in the x direction (forward).
+   * @param ySpeed        Speed of the robot in the y direction (sideways).
+   * @param rot           Angular rate of the robot.
+   * @param fieldRelative Whether the provided x and y speeds are relative to the
+   *                      field.
+   */
+  public void drive(double xSpeed, double ySpeed, double xRot, double yRot, boolean altDrive, boolean centerGyro) {
+    if(centerGyro) zeroHeading();
+    if(altDrive) {
+      altDrive(xSpeed, ySpeed, xRot, yRot);
+    } else {
+      mainDrive(xSpeed, ySpeed, xRot);
+    }
+  }
+
+  public void mainDrive(double xSpeed, double ySpeed, double xRot) {
+    double rot = xRot * DriveConstants.kMaxAngularSpeed;
+    move(xSpeed, ySpeed, rot);
+  }
+
+  public void altDrive(double xSpeed, double ySpeed, double xRot, double yRot) {
+    double rot = 0;
+    rightAngGoal = Math.atan2(xRot, yRot) * 180 / Math.PI;//convert to degrees
+    if(xRot != 0 || yRot != 0) {
+      double stickAng = Math.atan2(xRot, yRot) * 180 / Math.PI;//convert to degrees
```

Comment:
nit: please put the comment on the line above, or else leave 2 spaces between the end of the line and the start of the comment. Please also add a space after the start of a comment